CRM/Utils/String.php

```diff
@@ -1035,30 +1035,32 @@ public static function parseOneOffStringThroughSmarty($templateString) {
     set_error_handler([$smarty, 'handleSmartyError'], E_USER_ERROR);
     $smarty->caching = 0;
     $smarty->assign('smartySingleUseString', $templateString);
-    // Do not escape the smartySingleUseString as that is our smarty template
-    // and is likely to contain html.
-    // The file name generated by
-    // 'string:{eval var=$smartySingleUseString|smarty:nodefaults}'
-    // is invalid in Windows, causing failure.
-    // Adding this is preparatory to smarty 3. The original PR failed some
-    // tests so we check for the function.
-    if (!function_exists('smarty_function_eval') && (!defined('SMARTY_DIR') || !file_exists(SMARTY_DIR . '/plugins/function.eval.php'))) {
-      try {
-        $templateString = (string) $smarty->fetch('eval:' . $templateString);
+    try {
+      // Do not escape the smartySingleUseString as that is our smarty template
+      // and is likely to contain html.
+      // The file name generated by
+      // 'string:{eval var=$smartySingleUseString|smarty:nodefaults}'
+      // is invalid in Windows, causing failure.
+      // Adding this is preparatory to smarty 3. The original PR failed some
+      // tests so we check for the function.
+      if (!function_exists('smarty_function_eval') && (!defined('SMARTY_DIR') || !file_exists(SMARTY_DIR . '/plugins/function.eval.php'))) {
+          $templateString = (string) $smarty->fetch('eval:' . $templateString);
       }
-      catch (SmartyCompilerException $e) {
-        \Civi::log('smarty')->info('parsing smarty template {template}', [
-          'template' => $templateString,
-        ]);
-        throw new \CRM_Core_Exception('Message was not parsed due to invalid smarty syntax : ' . $e->getMessage() . ((CIVICRM_UF === 'UnitTest' || CRM_Utils_Constant::value('SMARTY_DEBUG_STRINGS')) ? $templateString : ''));
+      else {
+        $templateString = (string) $smarty->fetch('string:{eval var=$smartySingleUseString|smarty:nodefaults}');
       }
     }
-    else {
-      $templateString = (string) $smarty->fetch('string:{eval var=$smartySingleUseString|smarty:nodefaults}');
+    catch (SmartyCompilerException $e) {
```

Comment:
I agree with the concept of adding "finally" but I think this exception doesn't exist in smarty2 which explains why it was in the `if`. Also it seemed to be throwing a RuntimeException in the test which was why it wasn't getting caught here.

Reply:
oh....

Reply:
what say we just catch Exception?

Reply:
Sounds good.